dali/operators/reader/loader/video/video_loader_decoder_gpu.cc

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/loader/video/video_loader_decoder_gpu.h"
+
+namespace dali {
+void VideoLoaderDecoderGpu::PrepareEmpty(VideoSample<GPUBackend> &sample) {
+  sample = {};
+}
+
+void VideoLoaderDecoderGpu::ReadSample(VideoSample<GPUBackend> &sample) {
+  auto &sample_span = sample_spans_[current_index_];
+  auto &video_file = video_files_[sample_span.video_idx_];
+
+  ++current_index_;
+  MoveToNextShard(current_index_);
+
+  TensorShape<4> sequence_shape = {
+    sequence_len_, video_file.Height(), video_file.Width(), video_file.Channels()};
+
+  sample.data_.Resize(
+    sequence_shape,
+    DALIDataType::DALI_UINT8);
+
+  auto data = sample.data_.mutable_data<uint8_t>();
+
+  // TODO(awolant): Extract decoding outside of ReadSample (ReaderDecoder abstraction)
+  for (int i = 0; i < sequence_len_; ++i) {
+    // TODO(awolant): This seek can be optimized - for consecutive frames not needed etc.
```

Comment:
Maybe we can optimize the seek itself and keep it here even with the optimization?

Reply:
I moved the comment to `Seek`. This will be done as DALI-2320.